### DISCUSSION TITLE
Bind file rename to F2 by default?

### DISCUSSION THREAD

**Discussion by slovuj** [sylikc/jpegview#120] [Ideas]
Given that 100% of the MS ecosystem uses F2 to rename files, dirs, edit Excel cells, etc.,
I customized my KeyMap.txt to:

```
F2                      IDM_RENAME
Ctrl+Shift+F2    IDM_SHOW_FILEINFO
Ctrl+F2             IDM_SHOW_FILENAME 

```
What do others think? I use the keyboard EXCLUSIVELY in all my apps, unless 
something can only be done with a mouse, renaming is no exception. Locating
and clicking the edit input box with the mouse is humiliating when you can just 
hit F2 and start typing blindly!

**Comment by sylikc** [comment]
I kinda agree with you... There's part of me that doesn't want to change the original author's defaults, and then there's that.

Like when I put in the new feature for hide title bar, I wanted to use `F12` because in Notepad++ that's what the key is.
`F2` is certainly universal for rename... there's a few others, like `F5` should be reload (or in my case, reload file list), etc etc... but I'm not sure which ones to change or not.

It's great that `KeyMap.txt` can be used to customize to your liking, but yeah, the defaults matter too I guess.

Thanks for the great suggestions, keep 'em coming.  I think while these are a bit backlog for me, they're worth thinking about.

**Reply by slovuj** [reply]
_Like when I put in the new feature for hide title bar, I wanted to use F12 because in Notepad++ that's what the key is._
My editor as well. But it's a pity it is not F11, because browsers use that to hide the title bar and the menus and bars. In browsers, F12 is for devs.
I can live with syncing my KeyMap.txt as needed, let's not disrupt current habits :-).
Plus, I'm sure you have higher priorities to work on.

**Reply by sylikc** [reply]
F11 is the standard for fullscreen, so that's great as it is.

It's still a good suggestion, I'll keep it in mind.  Thanks

**Reply by pureby** [reply]
> I kinda agree with you... There's part of me that doesn't want to change the original author's defaults, and then there's that.

I totally get that. All I can say is: it's now your project, your vision and your community ;-)

My personal opinion is that it's OK to change defaults, if it enhances usability and results in overall more universal and streamlined experience for the end user. Software development has gone a long way since 2018, there is nothing bad in letting old habits go in favor of new and improved ones.

**Reply by pureby** [reply]
A thought occurred, just a simple mind-experiment: _if you were ever to port JPEGView to Mac, would you keep all the default Windows behaviors, or would you try to change them to those which most closely resemble the native workflow of a Mac user?_

The same question can be asked about porting a 2018-Windows experience to 2023-Windows experience.

**Reply by slovuj** [reply]
Slightly off-topic: Notepad++ has two full screen modes:
`F12 `- removes title, menus, tabs with open files, all that is left is pure client area with text. 
So, F12 is the "pure" full screen, the way it acts when F11 is hit in browsers and other apps.

`F11` - does the same, except that it **keeps the tabs ribbon visible**, so you can see/switch tabs
more easily. I think these should be swapped for consistency, but we are talking about
millions of user of that editor. Too late.

In JPEGview, we may want to think about assigning F11 to its current full screen function. If one day
we decide we need to keep some info still visible, F12 would be the key for this "impure"
full screen.

**Comment by sylikc** [comment]
> Given that 100% of the MS ecosystem uses F2 to rename files, dirs, edit Excel cells, etc., I customized my KeyMap.txt to:
> 
> ```
> F2                      IDM_RENAME
> Ctrl+Shift+F2    IDM_SHOW_FILEINFO
> Ctrl+F2             IDM_SHOW_FILENAME 
> ```
> 
> What do others think? I use the keyboard EXCLUSIVELY in all my apps, unless something can only be done with a mouse, renaming is no exception. Locating and clicking the edit input box with the mouse is humiliating when you can just hit F2 and start typing blindly!

I use the keyboard for apps almost exclusively too... except in JPEGView lol.  Anyways, this is a good suggestion, and I'll probably end up making another set of defaults.

Locking this discussion... will capture this in #140 .  All further discussion will be grouped to that one issue.
